### PR TITLE
1.1.0

### PR DESCRIPTION
**Pull request by robertbastian**
## icu_calendar
* Fix bug in simple_week_of() around unit size (#2951)
* Fix math in calendar (#2714)
* Add div_rem_euclid and use it in icu_calendar (#2704)
* Fix Time::from_minute_with_remainder_days to handle negatives (#2643) (#2702)
* doc improvements

## icu_casemapping
* doc improvements

## icu_collator
* doc improvements

## icu_collections
* Add APIs for returning exemplar characters data (#2812)
* Readable JSON inversion lists (#2855)
* Add UnicodeSet that supports strings (#2796)
* Add documentation on CodePointTrie details and perf considerations (#2717)

## icu_codepointtrie_builder
* internal improvements

## icu_datetime
* internal and doc improvements

## icu_decimal
* internal and doc improvements

## icu_displaynames
* DisplayNames fixes (#2918)
* Rename Territory -> Region for Displayname component (#2895)
* Transformer code for Language display names (#2871)
* Adding a function to get display-name for a region. (#2816)
* Transformer code for DisplayNames component. (#2635)
* doc improvements

## icu_list
* Untangling list provider from logic and fixing big endian safety bug (#2994)
* Not allocating Writeables for regex evaluation (#2991)
* doc improvements

## icu_locid
* Reject duplicated extensions (#2893)
* More borrowing in locid's `write_to_string` (#2693)
* doc improvements

## icu_locid_transform
* Clean up dependency specifications so serde isn't pulled in by default (#2696)
* doc improvements

## icu_normalizer
* internal and doc improvements

## icu_plurals
* doc improvements

## icu_properties
* Add APIs for returning exemplar characters data (#2812)
* Add API and testdata for Basic_Emoji property (#2802)
* Add UnicodeSet that supports strings (#2796)
* Update Script property value enums (#2787)
* doc improvements

## icu_segmenter
* Make metacrate features more specific (#2932)
* Remove serde dependency from segmenter with lstm feature. (#2904)
* Simplify construction of grapheme cluster break iterators (#2870)
* Store grapheme cluster payload instead of grapheme cluster segmenter. (#2864)
* `#[no_std]` for LSTM segmenter (#2845)
* icu_segmenter: enforce clippy::indexing_slicing. (#2325)
* Use GraphemeClusterSegmenter in DictionarySegmenter and LstmSegmenter (#2716)
* Rename *BreakSegmenter to *Segmenter (#2707)
* Remove unnecessary langauge check for East Asian languagne (SA property) (#2705)
* internal and doc improvements

## icu_timezone
* Adds a bytes parsing API for GMTOffset for CustomTimeZone and FFI (#2943, #2955)
* doc improvements

## icu_provider_adapters
* Add more inner_mut functions in icu_provider_adapters (#2987)
* Fix error propagation in MultiForkByErrorProvider (#2986)
* Add mutation methods to MultiForkByErrorProvider (#2972)

## icu_provider_blob
* internal and doc improvements

## icu_provider
* Setting correct DataError for .as_deserializing(), .as_downcasting() (#2993)
* doc improvements

## icu_datagen
* Removing experimental feature from datagen (#3005)
* Fixing Spanish list regex (#2989)
* Datagen CLI improvements (#2950)
* Some reexports for datagen (#2958)
* Databake improvements (#2906)
* Exclude certain collations by default and add option to include them (#2789)
* Allowing no keys in datagen CLI (#2731)
* Fixing baked datagen for no keys and keys with no data (#2698)
* Internal and doc improvements

## icu_provider_fs
* internal and doc improvements

## icu_testdata
* stabilized metadata
* doc improvements
* CLDR 42 and ICU 72 data

## bies
* Bumped writeable version

## databake
* Doc improvements

## fixed_decimal
* Treat strings starting with a decimal point as valid in FixedDecimal::from_str() (#2937)
* The actual formatting part of compact decimal formatting (#2898)
* Allow noncompact CompactDecimal (#2889)
* CompactDecimal and ScientificDecimal (#2847)
* doc improvements

## icu_pattern
* internal improvements

## litemap
* internal and doc improvements

## tinystr
* Add std feature and Error impl for TinyStrError (#3009)
* internal and doc improvements

## tzif
* internal and doc improvements

## writeable
* Using core integer log when available (#3015)
* `usize` and `isize` implementation
* internal and doc improvements

## yoke
* Add `prove_covariance_manually` guard for `CoerceUnsized` (#2936)
* Allow `clippy::forget_copy` in `derive(Yokeable)` impl (#2775)
* Fix soundness issue in `Yoke::attach_to_cart()` around implied bounds #2949

## zerovec
* ZeroMap2d cursors

### CODE REVIEW

**Comment by Manishearth** [comment]
@robertbastian @sffc can we land https://github.com/unicode-org/icu4x/pull/2949 and bump Yoke first? it's technically a breaking change.

I'd rather not block that on IsCovariant, a thing which is currently unused and I'd like to redesign again based on actual need.

**Comment by robertbastian** [comment]
I haven't seen any activity on that PR, and it's not in the 1.1 Blocking milestone. What's the ETA on it?

**Comment by Manishearth** [comment]
@robertbastian it should be mergeable once Shane responds

it didn't occur to me that it should be blocking, but i think it should

**Comment by Manishearth** [comment]
@robertbastian feel free to roll https://github.com/unicode-org/icu4x/pull/3019 into your PR, but I can just merge and publish independently

**Comment by Manishearth** [comment]
Also, changelog for Yoke

 - Add `prove_covariance_manually` guard for `CoerceUnsized` (#2936)
 - Allow `clippy::forget_copy` in `derive(Yokeable)` impl (#2775)
 - Fix soundness issue in `Yoke::attach_to_cart()` around implied bounds #2949

File: components/calendar/Cargo.toml [hunk 1]
```diff
@@ -28,13 +28,13 @@ all-features = true
 
 [dependencies]
 displaydoc = { version = "0.2.3", default-features = false }
-icu_provider = { version = "1.0.0", path = "../../provider/core", features = ["macros"] }
-icu_locid = { version = "1.0.0", path = "../../components/locid" }
-tinystr = { version = "0.7", path = "../../utils/tinystr", features = ["alloc", "zerovec"], default-features = false }
-zerovec = { version = "0.9", path = "../../utils/zerovec", default-features = false, features = ["derive"] }
-writeable = { version = "0.5", path = "../../utils/writeable" }
+icu_provider = { path = "../../provider/core", features = ["macros"] }
```
Comment (sffc) [inline components/calendar/Cargo.toml:31]: Question: How are you able to remove these versions? I think `cargo publish` doesn't let you publish unless all of your `dependencies` have a version. Is this `cargo workspaces publish` magic?

File: experimental/segmenter/Cargo.toml [hunk 2]
```diff
@@ -24,18 +24,21 @@ include = [
     "README.md"
 ]
 
+[package.metadata.workspaces]
+independent = true
```
Comment (sffc) [inline experimental/segmenter/Cargo.toml:28]: I'm not really sure what are the implications of this line

**Review by sffc — COMMENTED** [review]
Haven't read up yet on `cargo workspaces` but it looks like it might be quite helpful

File: components/calendar/Cargo.toml (hunk 1 above)
Comment (robertbastian) [reply components/calendar/Cargo.toml:31]: I *think* `cargo workspaces` allows this. I'll know when I try.

File: experimental/segmenter/Cargo.toml (hunk 2 above)
Comment (robertbastian) [reply experimental/segmenter/Cargo.toml:28]: `cargo ws version` can be used to bump the versions. By default all crates are versioned together.

**Comment by Manishearth** [comment]
Hmm, won't `cargo ws` prevent us from doing out-of-cycle dot releases since it wants things to be in sync? I'm perhaps not fully clear on what the tool _does_ (at a lower level).

**Comment by Manishearth** [comment]
Can we use Cargo's regular [workspace-level versioning support](https://doc.rust-lang.org/cargo/reference/workspaces.html#the-dependencies-table)? It's not available on our MSRV, but that doesn't actually matter since _local_ tests won't hit it (I _think_?), and `cargo` normalizes before publish so crates.io users don't need to worry.

I'd like to stay as close to vanilla Cargo as possible. I'm worried about introducing a dependency on a new tool which is only invoked once in a blue moon (making it much harder to catch any bugs). We have problems with `cargo make` at times but they're caught immediately, which makes it tolerable, I'm worried about 

And plus; publishes aren't a retryable operation; if a publish succeeds but was run with the wrong options, you can't really undo that. I'm still unclear on how this tool works but it ... might be doing some Cargo.toml fixups before publishing, which seems super brittle.

I'm fine with using this tool for bumping stuff, but _highly_ suspicious about requiring it for publishing.

**Comment by robertbastian** [comment]
I was expected `cargo ws` to set the versions, but apparently it doesn't. Are we sure that `cargo publish` doesn't work with just a path? The path fully determines the dependency's version, because it points at a local crate with one version. I was hoping to get rid of all the duplicated versions.

I don't think we can use "workspace-level versioning support" because then our MSRV CI would fail (although I guess we could do something like `cargo pacakage` and then run MSRV on the result).

**Comment by sffc** [comment]
`cargo publish` has definitely complained when I've tried in the past if a dependency didn't have a version number

I'm pretty sure `cargo check` will complain loudly if one project's version number is out of sync with a dependent's Cargo.toml entry.

The well-trodden path that has worked for us in previous releases is to just go ahead and update all the version numbers and then `cargo publish` in topological order

**Comment by robertbastian** [comment]
> The well-trodden path that has worked for us in previous releases is to just go ahead and update all the version numbers and then cargo publish in topological order

I was hoping to improve on that because that's a lot of stupid work, but alas not.

File: experimental/segmenter/Cargo.toml (hunk 2 above)
Comment (sffc) [reply experimental/segmenter/Cargo.toml:28]: Should we remove these `[package.metadata.workspaces]` lines now?

**Review by sffc — APPROVED** [review]

**Comment by sffc** [comment]
Also, if you discover any further issues while publishing (not unlikely), so long as they are only Cargo.toml changes, I've usually pushed them directly to the main branch and asked for a post-submit review

File: experimental/segmenter/Cargo.toml (hunk 2 above)
Comment (robertbastian) [reply experimental/segmenter/Cargo.toml:28]: I'd still suggest `cargo ws version` to bump version, it's really convenient.

**Comment by robertbastian** [comment]
I'm not sure I can push to main.

**Pull request merged**